extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcUtils.java

```diff
@@ -532,6 +532,24 @@ public static String decryptString(String jweString, Key key) throws Exception {
         return decryptString(jweString, key, KeyEncryptionAlgorithm.A256GCMKW);
     }
 
+    public static String decryptToken(String token, RoutingContext context) {
```

Comment:
@michalvavrik  I'm not sure the db state manager needs to worry about encrypting/decrypting - it is only relevant for the situation, where we have tokens in the cookie, and since the cookie, with this option, will just have a DB pointer, we don't really need to encrypt/decrypt then before saving in the database or encrypting decrypting this cookie value.

Reply:
@sberyozkin hey Sergey, I think there is possible compromise - I'll change PK to database generated UUID. I'll do it tonight. This way, no encryption is needed, while it's still safe.

Reply:
@sberyozkin please hold with the Quarkus-dev e-mail till I fix that tonight. thank you!

Reply:
pity I didn't realize this before :-)

Reply:
>in case they have direct access to database

Then it is already all has gone wrong, we can encrypt as much as we want but they will just access the database.

>then next cookie value for other user can't be guessed!

Sure, I thought it was a random value, each new record key, as opposed to some predictable increment. But even if they guess the next value, what is the risk  ?

How would it be different, conceptually, from the traditional stateful applications where the cookie is essentially a pointer to some server side state ?

That said, indeed, lets keep it as you have thought about it for now, and discuss a long the way

Thanks

Reply:
@sberyozkin I'm going to drop encryption and use UUID instead

Reply:
> Then it is already all has gone wrong, we can encrypt as much as we want but they will just access the database.

I had in my mind DBA that can access DB, but not app secret etc. Anyway, let's go with DB-generated UUID.

Reply:
> Sure, I thought it was a random value, each new record key, as opposed to some predictable increment. But even if they guess the next value, what is the risk ?

I would need to spend more time on this, but I was afraid that someone will be authenticated, but will have access to someone else tokens pointer. Maybe that's rubbish, it is better to make it random.